pkg/frontend/transport/handler_test.go

```diff
@@ -267,6 +271,90 @@ func TestHandler_FailedRoundTrip(t *testing.T) {
 	}
 }
 
+// Test Handler.Stop.
+func TestHandler_Stop(t *testing.T) {
+	// Test that Stop waits for in-flight queries to finish.
+	t.Run("in-flight queries", func(t *testing.T) {
```

Comment:
Q: There is only one test, so why run it in a subtest?

Reply:
@pstibrany I was thinking that this is just one scenario for `Handler.Stop`. With sub-tests, we can keep adding more. Make sense?

Reply:
Having chatted with Peter offline, I'll drop the sub-test.